doc/templates/board.tmpl

```diff
@@ -1,20 +1,15 @@
-.. _boardname_linkname:
+.. zephyr:board:: board_name
 
-[Board Name]
-#############
+.. In order for the board documentation page to show properly in the
```

Comment:
@kartben overall LGTM, I would just improve this text to:
"To ensure the board documentation page displays correctly, it is highly recommended to include a picture of the board."